Bucketeer/Sources/Internal/Evaluation/EvaluationStatesStorage.swift

```diff
@@ -0,0 +1,16 @@
+import Foundation
+
+protocol EvaluationStateStorage {
```

Comment:
It might be better if we implement this in the `EvaluationDao`. Because both are storing data.

Reply:
There are some differences between Android and iOS SDK implementation compared to the JS implementation. Let me provide some context to help you understand my decision.



### JS SDK:
In the JS SDK, we have a well-designed interface called "BKTEvaluationStorage."
- This interface allows us to access all data related to evaluation.
- The great thing is that we can use various implementations of "BKTStorage," such as local databases or in-memory storage.
- The "EvaluationInteractor" utilizes the "BKTEvaluationStorage" for its functionality.



### Mobile SDK:
On the mobile side, the "EvaluationInteractor" uses "EvaluationDao" for abstract SQLLite database access, and it also uses a Map for in-memory caching. Additionally, it directly accesses iOS "UserDefault."
- "EvaluationDao" is not just for CURD operation, it contains business logic on the the method deleteAllAndInsert()



Considering the differences between the implementations in JS and the Mobile SDKs, I thought about creating similar implementations like "BKTEvaluationStorage", "BKTStorage" and "BKTEventStorage" in JS. However, this would require significant changes to the source code of the mobile SDK, affecting both implementation and testing.



To keep things simple and ensure a single responsibility, I've decided to stick with the existing "EvaluationDao" for database access (using SQLLite). However, I've introduced a new class called "EvaluationStateStorage." This class serves as a way to access simple state data that can be cached in iOS "UserDefault." If you think the name "EvaluationStateStorage" doesn't sound appropriate, please feel free to suggest an alternative name

Reply:
I changed `EvaluationStatesStorage` to `UserEvaluationStorage` 

`EvaluationStorage` will make it confused with the `EvaluationDao` 

 `UserEvaluationStorage` will provide access to `UserEvaluation`  refs : https://github.com/bucketeer-io/javascript-client-sdk/blob/a30fdc6acace764a366e5a0b98b51bf036463b51/src/internal/evaluation/EvaluationStorage.ts#L4
 
`EvaluationDao` will provide access to `Evaluation` like before

Reply:
JS SDK and Mobile SDKs have different platform restrictions.

Browser only has localStorage API, and it's fast enough to access directly. So we don't need to cache values in memory.

On the other hand, Mobile SDKs have a full-fledged SQL database, but it's slow and needs an in-memory cache.

One design mistake here is that we don't abstract the in-memory cache and database. 

So I think it's fine to create something like `EvaluationStorage` in JS SDK, but then it should be an abstraction layer over the dao, in-memory cache and UserDefaults.

Reply:
hi @yshrsmz  thank you for explaining the difference.
Actually, I am refactoring the evaluation data access layer of the iOS SDK. Follow up on your best practice on the JS SDK.
I also created something like your suggestion above.
I will do that for the Android Kotlin SDK to keep the consistency across all SDK.

cc: @cre8ivejp